solidity/test/bridge/Bridge.MovingFunds.test.ts

```diff
@@ -1929,42 +1996,1104 @@ describe("Bridge - Moving funds", () => {
+
+                                before(async () => {
+                                  await createSnapshot()
+                                })
+
+                                after(async () => {
+                                  await restoreSnapshot()
+                                })
+
+                                it("should revert", async () => {
+                                  // However, register a main UTXO for the
+                                  // sweeping wallet in order to force the
+                                  // system to expect a two-input transaction
+                                  // for that sweeping wallet.
+                                  await expect(
+                                    runMovedFundsSweepScenario({
+                                      ...data,
+                                      // Just an arbitrary main UTXO
+                                      mainUtxo: {
+                                        txHash:
+                                          "0x7d5f7d4ae705d6adb8a402e5cd7f25f839a3f3ed243a8961c8ac5887d5aaf528",
+                                        txOutputIndex: 0,
+                                        txOutputValue: 873510,
+                                      },
+                                    })
+                                  ).to.be.revertedWith(
+                                    "Moved funds sweep transaction must have a proper inputs count"
+                                  )
+                                })
+                              }
+                            )
+                          }
+                        )
+                      }
+                    )
+
+                    context(
+                      "when transaction fee exceeds the sweep transaction maximum fee",
+                      () => {
+                        // Use a test data where the sweep transaction has
+                        // a fee of 2000 satoshi.
+                        const data: MovedFundsSweepTestData =
+                          MovedFundsSweepWithoutMainUtxo
+
+                        before(async () => {
+                          await createSnapshot()
+
+                          // Set the max fee to one satoshi less than the fee
+                          // used by the transaction.
+                          await bridge.setMovedFundsSweepTxMaxTotalFee(1999)
+                        })
+
+                        after(async () => {
+                          await restoreSnapshot()
+                        })
+
+                        it("should revert", async () => {
+                          await expect(
+                            runMovedFundsSweepScenario(data)
+                          ).to.be.revertedWith("Transaction fee is too high")
+                        })
+                      }
+                    )
+                  })
+
+                  context("when main UTXO data are invalid", () => {
```

Comment:
Wouldn't it be simpler to use `runMovedFundsSweepScenario` with `BridgeStub.setWalletMainUtxo` called in `beforeProofActions`?

Reply:
Indeed! See https://github.com/keep-network/tbtc-v2/pull/244/commits/df39c2aa777925e39daf81f23a30d1e19781450c